gapil/runtime/cc/map.inc

```diff
@@ -98,13 +103,84 @@ inline void Map<K, V>::Allocation::release() {
     arena->free(this);
 }
 
-template<typename K, typename V>
-bool Map<K, V>::Allocation::contains(K key) {
+template<typename K, typename V, bool DENSE>
+bool Map<K, V, DENSE>::Allocation::contains(K key) {
     return index(key, false) != nullptr;
 }
 
-template<typename K, typename V>
-V* Map<K, V>::Allocation::index(K key, bool insert) {
+template<typename T, bool DENSE>
+typename std::enable_if<std::is_integral<T>::value, uint64_t>::type dense_hash(const T& t) {
+    return static_cast<uint64_t>(t);
+}
+
+
+template<typename T, bool DENSE>
+typename std::enable_if<!std::is_integral<T>::value, uint64_t>::type dense_hash(const T& t) {
+    return 0;
```

Comment:
Can we delete this implementation so that it won't compile if the key is non-integral (that or put some statement in here that doesn't compile with a comment)? Doesn't seem like you'd ever want this to compile...

Reply:
We can once 
    // TODO(awoloszyn): Template tricks to move this to compile-time
is resolved. Until then we have a dynamic check, which means this will need to compile.